src/utils/modules.js

```diff
@@ -208,95 +144,85 @@ export class ModuleTracer extends Tracer {
-          if (previousDependencies) {
-            // Verify that they are the same
-            let logError = () => {
-              let moduleId = argumentsList[1];
-              let moduleString =
-                moduleId instanceof StringValue || moduleId instanceof NumberValue ? moduleId.value : "unknown";
-              this.modules.logger.logError(
-                factoryFunction,
-                `Called define on the same module ${moduleString} twice with different dependencies each time.`
-              );
-            };
-            if (previousDependencies.length !== dependencies.length) {
-              logError();
-            } else {
-              let previousDependenciesSet = new Set(previousDependencies);
-              dependencies.forEach(dependency => {
-                if (!previousDependenciesSet.has(dependency)) logError();
-              });
-            }
-          } else {
+      let moduleId = argumentsList[1];
+      if (moduleId instanceof NumberValue || moduleId instanceof StringValue) {
+        let moduleIdValue = moduleId.value;
+        let factoryFunction = argumentsList[0];
+        if (factoryFunction instanceof FunctionValue) {
+          let dependencies = this._tryExtractDependencies(argumentsList[2]);
+          if (dependencies !== undefined) {
+            this.modules.moduleDependencies.set(moduleIdValue, dependencies);
             this.modules.factoryFunctionDependencies.set(factoryFunction, dependencies);
-          }
+          } else
+            this.modules.logger.logError(
+              argumentsList[2],
+              "Third argument to define function is present but not a concrete array."
+            );
         } else
-          this.modules.logger.logError(
-            argumentsList[2],
-            "Third argument to define function is present but not a concrete array."
-          );
+          this.modules.logger.logError(factoryFunction, "First argument to define function is not a function value.");
+
+        this.modules.moduleIds.add(moduleIdValue);
       } else
-        this.modules.logger.logError(factoryFunction, "First argument to define function is not a function value.");
-      let moduleId = argumentsList[1];
-      if (moduleId instanceof NumberValue || moduleId instanceof StringValue)
-        this.modules.moduleIds.add(moduleId.value);
-      else
         this.modules.logger.logError(moduleId, "Second argument to define function is not a number or string value.");
     }
     return undefined;
   }
 }
 
+export type RequireInfo = {
+  value: FunctionValue,
+  globalName: string,
+};
+
 export class Modules {
-  constructor(realm: Realm, logger: Logger, logModules: boolean, accelerateUnsupportedRequires: boolean) {
+  constructor(realm: Realm, logger: Logger, logModules: boolean) {
     this.realm = realm;
     this.logger = logger;
-    this._require = realm.intrinsics.undefined;
+    this._requireInfo = undefined;
```

Comment:
Since the default value is undefined, this is unnecessary and somewhat confusing.